consensus/src/consensus_observer/observer/consensus_observer.rs

```diff
@@ -271,6 +271,25 @@ impl ConsensusObserver {
             ))
         );
 
+        if self.pipeline_enabled() {
+            for block in ordered_block.blocks() {
+                let commit_callback = self.active_observer_state.create_commit_callback(
+                    self.ordered_block_store.clone(),
+                    self.block_payload_store.clone(),
+                );
+                if let Some(futs) = self.get_last_pipeline_futs() {
+                    self.pipeline_builder().build(block, futs, commit_callback);
+                } else {
+                    warn!(
+                            LogSchema::new(LogEntry::ConsensusObserver).message(&format!(
+                                "Parent block's pipeline futures for ordered block is missing! Ignoring: {:?}",
+                                ordered_block.proof_block_info()
+                            ))
+                        );
+                    return;
+                }
+            }
```

Comment:
@JoshLind there should be a return here iiuc? Otherwise, we are executing twice.

Reply:
IIUC and this orders the blocks for execution, then yeah, we should return, otherwise, we'll call `finalize_order` below and that will execute as well? 🤔

Reply:
we send to buffer manager regardless of new pipeline enabled or not, it still relies on buffer manager to aggregate votes/cert. buffer manager inspects the pipelined block to know whether it's enabled or not